app/src/main/java/com/randomappsinc/studentpicker/common/VoiceRecognizerDialogFragment.java

```diff
@@ -0,0 +1,161 @@
+package com.randomappsinc.studentpicker.common;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.speech.RecognitionListener;
+import android.speech.RecognizerIntent;
+import android.speech.SpeechRecognizer;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+
+import com.joanzapata.iconify.widget.IconTextView;
+import com.randomappsinc.studentpicker.R;
+
+import java.util.ArrayList;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+public class VoiceRecognizerDialogFragment extends DialogFragment implements RecognitionListener {
+
+    public interface VoiceRecognizerInterface {
+        void spokenText(String spokenText);
+    }
+
+    @BindView(R.id.voice_icon) IconTextView voiceIcon;
+    @BindView(R.id.message) IconTextView message;
+    @BindView(R.id.try_again) TextView tryAgain;
+
+    private SpeechRecognizer mSpeechRecognizer;
+    private Intent mSpeechRecognizerIntent;
+    private Context context;
+    private VoiceRecognizerInterface signal;
+
+    public VoiceRecognizerDialogFragment(Context context, VoiceRecognizerInterface signal) {
+        this.context = context;
+        this.signal = signal;
+    }
+
+    public VoiceRecognizerDialogFragment() {
+    }
+
+    @Override
+    public void onResume() {
```

Comment:
Be careful firing critical blocks in onResume(). It fires extremely often (instantiation, app background and return, return from activity on top, etc). In this case, this couples the speech recognition with the UI lifecycle, which makes the speech recognition kind of a side effect. With this design, the consumer of this module isn't really starting speech recognition, it's throwing up a fragment and the speech recognition just sort of happens. We should expose a more declarative API like startSpeechToText().